pwa/src/templates/templateParts/download/DownloadTemplate.tsx

```diff
@@ -1,63 +1,80 @@
 import * as styles from "./DownloadTemplate.module.css";
-
+import clsx from "clsx";
 import { DownloadCard, NotificationPopUp as _NotificationPopUp } from "@conduction/components";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faArrowLeft, faDownload } from "@fortawesome/free-solid-svg-icons";
 import { useTranslation } from "react-i18next";
-import clsx from "clsx";
 import { useGatsbyContext } from "../../../context/gatsby";
 
 interface DownloadTemplateProps {
-  icon: JSX.Element;
-  label: string;
-  sizeKb: string;
+  items: DownloadProps[];
+  backUrl: string;
+  icon?: JSX.Element;
   layoutClassName?: string;
 }
 
-export const DownloadTemplate: React.FC<DownloadTemplateProps> = ({ layoutClassName, icon, label, sizeKb }) => {
+export interface DownloadProps {
+  label: string;
+  type: string;
+  size: string;
+  downloadLink: string;
+}
+
+export const DownloadTemplate: React.FC<DownloadTemplateProps> = ({ items, backUrl, icon, layoutClassName }) => {
   const { t } = useTranslation();
   const { screenSize } = useGatsbyContext();
+  const [isVisible, setIsVisible] = React.useState<any>({});
 
-  const NotificationPopUpController = _NotificationPopUp.controller;
-  const NotificationPopUp = _NotificationPopUp.NotificationPopUp;
+  const toggleVisibility = (label: string) => {
+    setIsVisible({
+      [label]: {
+        value: true,
+      },
+    });
+  };
 
-  const { isVisible, show, hide } = NotificationPopUpController();
+  const NotificationPopUp = _NotificationPopUp.NotificationPopUp;
 
   return (
     <div className={clsx([layoutClassName && layoutClassName], screenSize === "mobile" && styles.downloadName)}>
-      <DownloadCard
-        label={label}
-        sizeKb={sizeKb}
-        downloadLabel="Download"
-        icon={icon}
-        handleClick={() => {
-          show();
-        }}
-      />
-
-      {isVisible && (
-        <div className={styles.overlay}>
-          <NotificationPopUp
-            {...{ hide, isVisible }}
-            title={`${t("Warning")}!`}
-            description={t(
-              "This file comes from a 3rd party and can potentially be harmfull for your PC. Are you sure you want to download this?",
-            )}
-            primaryButton={{
-              label: "Download",
-              icon: <FontAwesomeIcon icon={faDownload} />,
-              handleClick: () => ({}),
-            }}
-            secondaryButton={{
-              label: t("Go back"),
-              icon: <FontAwesomeIcon icon={faArrowLeft} />,
-              handleClick: () => ({}),
+      {items.map(({ label, size, type, downloadLink }, idx) => (
+        <React.Fragment key={idx}>
+          <DownloadCard
+            label={label ?? downloadLink}
+            type={type ?? downloadLink.substring(downloadLink.lastIndexOf(".") + 1)}
+            {...{ size, icon }}
+            handleClick={() => {
+              toggleVisibility(label ?? downloadLink);
```

Comment:
I'm unsure if defaulting to the `downloadLink` is the right thing to do for a label, see screenshot. Is there any other way we can actually find the file name perhaps? This file in particular is called `ZGWBundle-main`, which would be a good name to use (maybe something the backend can implement).

Otherwise, I think simply "No name available" might be better than showing the URL. 

![image](https://github.com/OpenCatalogi/web-app/assets/66552932/788bcf48-4ab5-415d-b736-407452cc6de5)